### PR TITLE
ENH: Add insert control point xyz function

### PR DESCRIPTION
**Pull request by sjh26**
Adds functions to insert a control point using a vector (previously only available using ControlPoint structure)

Also has a bugfix for resetting the ID on a control point.  Previously, LastUsedControlPointNumber was no incremented fro each reset, and so if you reset multiple points that would each gave the same ID.

### CODE REVIEW

File: Modules/Loadable/Markups/MRML/vtkMRMLMarkupsNode.cxx [hunk 1]
```diff
@@ -1414,7 +1435,9 @@ bool vtkMRMLMarkupsNode::ResetNthControlPointID(int n)
     {
     return false;
     }
+
   this->SetNthControlPointID(n, this->GenerateUniqueControlPointID());
+  this->LastUsedControlPointNumber++;
```
Comment (lassoan) [inline Modules/Loadable/Markups/MRML/vtkMRMLMarkupsNode.cxx]: This manual incrementing of LastUsedControlPointNumber is quite messy. I think it should be done in GenerateUniqueControlPointID() method.

Comment (jcfr) [reply Modules/Loadable/Markups/MRML/vtkMRMLMarkupsNode.cxx]: I also thought about this, but `LastUsedControlPointNumber` is not systematically incremented each time GenerateUniqueControlPointID is called ...

Comment (sjh26) [reply Modules/Loadable/Markups/MRML/vtkMRMLMarkupsNode.cxx]: This case is the only one where it (currently) isn't incremented?
https://github.com/Slicer/Slicer/blob/d3bd8dd0377fd437ad87f0cd6b288c57222571d3/Modules/Loadable/Markups/MRML/vtkMRMLMarkupsNode.cxx#L482

Is this correct or a bug?

**Comment by lassoan** [comment]
Merged (and cleaned up LastUsedControlPointNumber increment in a follow-up commit).

**Pull request closed without merging**